sdk/src/main/java/io/dapr/actors/runtime/ActorRuntime.java

```diff
@@ -0,0 +1,163 @@
+     * @param actorId Actor id for the actor to be activated.
+     */
+    static void Activate(String actorTypeName, String actorId)
+    {
+        // uncomment when ActorManager implemented
+        // return instance.GetActorManager(actorTypeName).ActivateActor(new ActorId(actorId));
+    }
+
+    /**
+     * Deactivates an actor for an actor type with given actor id.
+     * @param actorTypeName Actor type name to deactivate the actor for.
+     * @param actorId Actor id for the actor to be deactivated.
+     */
+    static void Deactivate(String actorTypeName, String actorId)
+    {
+        // uncomment when ActorManager implemented
+        // return instance.GetActorManager(actorTypeName).DeactivateActor(new ActorId(actorId));
+    }
+
+    /**
+     *  Invokes the specified method for the actor, this is mainly used for cross language invocation.
+     * @param actorTypeName Actor type name to invoke the method for.
+     * @param actorId Actor id for the actor for which method will be invoked.
+     * @param actorMethodName Method name on actor type which will be invoked.
+     * @param requestBodyStream Payload for the actor method.
+     * @param responseBodyStream Response for the actor method.
+     * @return
+     */
+    static void Dispatch(String actorTypeName, String actorId, String actorMethodName, byte[] requestBodyStream, byte[] responseBodyStream)
+    {
+        // uncomment when ActorManager implemented
+        // return instance.GetActorManager(actorTypeName).Dispatch(new ActorId(actorId), actorMethodName, requestBodyStream, responseBodyStream);
+    }
+
+    /**
+     * Fires a reminder for the Actor.
+     * @param actorTypeName Actor type name to invoke the method for.
+     * @param actorId Actor id for the actor for which method will be invoked.
+     * @param reminderName The name of reminder provided during registration.
+     * @param requestBodyStream Payload for the actor method
+     */
+    static void FireReminder(String actorTypeName, String actorId, String reminderName, byte[] requestBodyStream)
+    {
+        // uncomment when ActorManager implemented
+        // return instance.GetActorManager(actorTypeName).FireReminder(new ActorId(actorId), reminderName, requestBodyStream);
+    }
+
+    /**
+     * Fires a timer for the Actor.
+     * @param actorTypeName Actor type name to invoke the method for.
+     * @param actorId Actor id for the actor for which method will be invoked.
+     * @param timerName The name of timer provided during registration.
+     */
+    static void FireTimer(String actorTypeName, String actorId, String timerName)
+    {
+        // uncomment when ActorManager implemented
+        // return instance.GetActorManager(actorTypeName).FireTimerAsync(new ActorId(actorId), timerName);
+    }
+
+    private ActorManager GetActorManager(String actorTypeName) throws IllegalStateException
+    {
+        ActorManager actorManager = this.actorManagers.get(actorTypeName);
```

Comment:
We probably need to synchronize this, not?

Reply:
Insertion into dictionary and fetching from this method will not happen concurrenctly. The dictionary will be populated at startup before Dapr can make calls into it.

Reply:
From my understanding, there is a Dapr endpoint upstream that will be invoking this down the stream. How can we guarantee there will be only one http call being processed?

For example, multiple clients might be activating the same or different actors, then these calls will all come here concurrently unless you lock at the level above.

Reply:
Multiple threads reading from it don't need synchronization for this dictionary. Each of the call will get their actor manager and work with it.
Calls for the same actor id cannot come here concurrently, actor runtime in Dapr guarantees that.
Items in this dictionary are added at startup only.

Reply:
I still not sure about this not getting concurrent calls. I will not block this PR. I will revisit this code once we have the SDK working end-to-end.